lib/rubocop/cop/cop_store.rb

```diff
@@ -0,0 +1,160 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    # Error raised when an unqualified identifier is used that could
+    # refer to two or more cops under different namespaces
+    class AmbiguousCopName < RuboCop::Error
+      MSG = 'Ambiguous cop name `%<name>s` used in %<origin>s needs namespace' \
+            ' qualifier. Did you mean %<options>s?'.freeze
+
+      def initialize(identifier, origin, namespaces)
+        options = namespaces.map { |namespace| "#{namespace}/#{identifier}" }
+        message = format(
+          MSG,
+          name: identifier,
+          origin: origin,
+          options: options.join(' or ')
+        )
+
+        super(message)
+      end
+    end
+
+    # Store for all cops with helper functions
+    class CopStore
+      def initialize(cops = [])
+        @cops      = []
+        @cop_names = Set.new
+        @cop_types = {}
+
+        cops.each { |cop| register(cop) }
+      end
+
+      def register(cop)
+        @cops << cop
+        @cop_names.add(cop.cop_name)
+        @cop_types[cop.cop_type] ||= namespace(cop.cop_name)
```

Comment:
We use namespace and type interchangeably it would seem. I'd just go with the term `department`.